recipe/meta.yaml

```
@@ -1,19 +1,19 @@
 {% set name = "nccl" %}
-{% set version = "2.22.3-1" %}
+{% set version = "2.23.4-1" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version|replace("-", ".") }}
 
 source:
   url: https://github.com/NVIDIA/nccl/archive/v{{ version }}.tar.gz
-  sha256: 45151629a9494460e73375281e8b0fe379141528879301899ece9b776faca024
+  sha256: c3c90ab2bffb99927700789fe6c1f8f620c8f29f4ed545d48e3e09f51931a0e5
```

The CI build ran into compilation issues:
```
In file included from include/comm.h:14,
                 from include/channel.h:9,
                 from channel.cc:7:
include/proxy.h:16:10: fatal error: shmutils.h: No such file or directory
   16 | #include "shmutils.h"
```

That issue appears to have been fixed: https://github.com/NVIDIA/nccl/issues/1448

However, the fix was released by moving the git tag (not making a new one): https://github.com/NVIDIA/nccl/issues/1448#issuecomment-2354686074

Therefore, this sha256sum is incorrect. The result of `wget -qO- https://github.com/NVIDIA/nccl/archive/v2.23.4-1.tar.gz | sha256sum` gives:
```suggestion
  sha256: 6b946b70a9d2d01871842cbd15ec56488d358abe9a0f3767e372fddc3e241ba7
```

Updating that SHA256 sum should rerun CI and fix the issue.